integrations/airflow/marquez_airflow/extractors/__init__.py

```diff
@@ -166,12 +176,11 @@ def validate(self):
         assert (self.operator_class is not None and
                 self.operator.__class__ == self.operator_class)
 
-    # TODO: Only return a single StepMetadata object.
     @abstractmethod
-    def extract(self) -> [StepMetadata]:
+    def extract(self) -> StepMetadata:
```

Comment:
the only thing is that we might break compatibility for existing custom extractors.

Reply:
(but otherwise this is a good change)

Reply:
Added few compatibility changes for this situation